app/controllers/search_controller.rb

```diff
@@ -4,20 +4,37 @@ class SearchController < SessionsController
   before_action :current_user
 
   def explore
-    @repositories = Repository.paginate(per_page: 12, page: params[:page]).public_repos.order([sort_order].to_h).page params[:page]
+    if params[:category].blank?
+      @repositories = Repository.paginate(per_page: 12, page: params[:page]).public_repos.order([sort_order].to_h).page params[:page]
+    else
+      categories = Category.where(name: SLUG_TO_CATEGORY_MODEL[params[:category]]).where.not(repository_id: nil).distinct.includes(:repository).map(&:repository)
+      @repositories = Repository.paginate(per_page: 12, page: params[:page]).public_repos.where(id: categories.map(&:id)).order([sort_order].to_h).page params[:page]
```

Comment:
I think it could be simplified that way (To be double checked)
```rb
@repositories = Repository.paginate(per_page: 12, page: params[:page]).public_repos.includes(:categories).where({categories: {name: params[:category]}}).order([sort_order].to_h).page params[:page]
```